juju/juju
# ec2: start instance in same zone as existing volume attachment(s)

## Description of change

Some providers, like AWS, require that VMs and attached disks/volumes be in the same availability zone. When deploying an application with --attach-storage, we should create the first unit's machine instance in the same availability zone as the existing volume.

## QA steps

1. juju bootstrap aws/ap-southeast-2
2. juju deploy postgresql --storage pgdata=ebs,10G --to zone=ap-southeast-2c
3. juju remove-application postgresql
4. juju deploy postgresql pg2 --attach-storage pgdata/0

pg2/0's machine should be in AZ ap-southeast-2c

## Documentation changes

Using "--to zone=..." will be incompatible with --attach-storage, unless you happen to specify the same zone as the volumes are in.

## Bug reference

None.
